### PR TITLE
Examples from ICFP paper merged into master 

### PR DESCRIPTION
**Pull request by michel-steuwer**

### CODE REVIEW

File: src/test/scala/apps/separableConvolution2DRewrite.scala [hunk 1]
```diff
@@ -168,6 +168,17 @@ class separableConvolution2DRewrite extends shine.test_util.Tests {
     ))
   }
 
+//  test("scanline to regRotSeq") {
```
Comment (bastian-koepcke) [inline src/test/scala/apps/separableConvolution2DRewrite.scala:171]: should this stay commented out?

File: src/main/scala/util/gen.scala [hunk 2]
```diff
@@ -14,15 +14,15 @@ object gen {
     val dpia_e = toDPIA(e)
     val p = shine.C.ProgramGenerator.makeCode(dpia_e, name)
     SyntaxChecker(p.code)
-    println(p.code)
+    //println(p.code)
```
Comment (bastian-koepcke) [inline src/main/scala/util/gen.scala:17]: can I remove these?

**Review by bastian-koepcke — DISMISSED** [review]

File: src/main/scala/util/gen.scala [hunk 3]
```diff
@@ -14,15 +14,15 @@ object gen {
...
     p
   }
 
   def OpenMPProgram(e: rise.core.Expr, name: String = "foo"): shine.OpenMP.Program = {
     val dpia_e = toDPIA(e)
     val p = shine.OpenMP.ProgramGenerator.makeCode(dpia_e, name)
     SyntaxChecker(p.code)
-    println(p.code)
+    //println(p.code)
```
Comment (Bastacyclop) [inline src/main/scala/util/gen.scala:25]: we can remove the comment?

File: src/test/scala/apps/separableConvolution2DRewrite.scala [hunk 4]
```diff
@@ -157,6 +157,17 @@ class separableConvolution2DRewrite extends shine.test_util.Tests {
...
+//    rewrite_steps(toTDSL(scanline)(weightsV)(weightsH), Seq(
+//      (repeatNTimes(2, topDown(lowering.reduceSeqUnroll)) `;`
+//        topDown(lowering.slideSeq(SlideSeq.Values, idE)) `;`
+//        BENF `;`
+//        topDown(algorithmic.slideSeqFusion) `;`
+//        topDown(lowering.mapSeq))
+//        -> toTDSL(regRotSeq)(weightsV)(weightsH)
+//    ))
+//  }
```
Comment (Bastacyclop) [inline src/test/scala/apps/separableConvolution2DRewrite.scala:169]: I think we can remove this, the test is already in the rest of the code.

File: src/test/scala/elevate/rise/tvmGemm.scala [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+          openMP: Boolean // generate C or OpenMP code?
+         ): Unit = {
+
+//    def toFile(path: String, name: String, content: String, ending: String = ".c"): Unit = {
+//      import java.io._
+//      val w =new PrintWriter(new File(s"$path/$name$ending"))
+//      w.write(content)
+//      w.flush()
+//      w.close()
+//    }
```
Comment (Bastacyclop) [inline src/test/scala/elevate/rise/tvmGemm.scala:161]: do we need to keep these commented out code chunks?

File: src/test/scala/shine/DPIA/Store.scala [hunk 6]
```diff
@@ -0,0 +1,51 @@
+package shine.DPIA
+
+import rise.core._
+import rise.core.TypedDSL._
+import rise.core.types._
+import util.gen
+
+class Store extends shine.test_util.Tests {
+
+  test("scalar values") {
```
Comment (Bastacyclop) [inline src/test/scala/shine/DPIA/Store.scala:10]: looks like a duplicate of another test https://github.com/rise-lang/shine/blob/master/src/test/scala/shine/DPIA/Primitives/Let.scala#L12 ? maybe we should only keep one.

**Review by Bastacyclop — CHANGES_REQUESTED** [review]

File: src/main/scala/util/gen.scala (hunk 3 above)
Comment (michel-steuwer) [reply src/main/scala/util/gen.scala:25]: I have kept the `println` uniformly for all versions.

File: src/test/scala/elevate/rise/tvmGemm.scala (hunk 5 above)
Comment (michel-steuwer) [reply src/test/scala/elevate/rise/tvmGemm.scala:161]: I have uncommented the code and made it an easy to opt-in option of the test to store the intermediate results on disc (as it was done for the ICFP 2020 artifact).

File: src/test/scala/apps/separableConvolution2DRewrite.scala (hunk 4 above)
Comment (michel-steuwer) [reply src/test/scala/apps/separableConvolution2DRewrite.scala:169]: Removed.

File: src/test/scala/shine/DPIA/Store.scala (hunk 6 above)
Comment (michel-steuwer) [reply src/test/scala/shine/DPIA/Store.scala:10]: I have removed the `Let` test.

**Review by Bastacyclop — APPROVED** [review]

**Pull request merged**